src/alfasim_sdk/_internal/alfacase/plugin_alfacase_to_case.py

```diff
@@ -0,0 +1,402 @@
+from functools import partial
+from pathlib import Path
+from pathlib import PurePosixPath
+from types import ModuleType
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+import attr
+from barril.units import Array
+from barril.units import Scalar
+from typing_extensions import TypeGuard
+
+from alfasim_sdk import BaseField
+from alfasim_sdk._internal.alfacase.alfacase_to_case import DescriptionDocument
+from alfasim_sdk._internal.alfacase.alfacase_to_case import load_value
+from alfasim_sdk._internal.alfacase.alfacase_to_case import to_case_values
+from alfasim_sdk._internal.alfacase.case_description import CaseDescription
+from alfasim_sdk._internal.alfacase.case_description import PluginDescription
+from alfasim_sdk._internal.alfacase.case_description import PluginFileContent
+from alfasim_sdk._internal.alfacase.case_description import PluginInternalReference
+from alfasim_sdk._internal.alfacase.case_description import PluginMultipleReference
+from alfasim_sdk._internal.alfacase.case_description import PluginTableContainer
+from alfasim_sdk._internal.alfacase.case_description import PluginTracerReference
+from alfasim_sdk._internal.alfacase.case_description_attributes import (
+    InvalidPluginDataError,
+)
+from alfasim_sdk._internal.alfacase.plugin_introspection import get_attributes
+from alfasim_sdk._internal.types import Boolean
+from alfasim_sdk._internal.types import Enum
+from alfasim_sdk._internal.types import FileContent
+from alfasim_sdk._internal.types import MultipleReference
+from alfasim_sdk._internal.types import Quantity
+from alfasim_sdk._internal.types import Reference
+from alfasim_sdk._internal.types import String
+from alfasim_sdk._internal.types import Table
+from alfasim_sdk._internal.types import TracerType
+
+_CHILDREN_LIST_KEY = "_children_list"
+
+
+def get_unused_filename(alfacase_folder: Path, path: Path) -> Path:
+    """
+    Return a not used file name inside ``alfacase_folder``.
+    """
+    n = 1
+    extra = ""
+    while True:
+        fullpath = alfacase_folder / f"{path.stem}{extra}{path.suffix}"
+        if not fullpath.exists():
+            return fullpath
+        else:
+            n += 1
+            extra = f" ({n})"
```

Comment:
```
 PurePath.with_stem(stem)
    ...
    New in version 3.9.
```
=/

Reply:
```python

    # The original file does not need to actually exist (`path.stem` could be empty).
    stem = path.stem or '(unnamed)'
    fullpath = alfacase_folder / f"{stem}{path.suffix}"
    for i in itertools.count(2):
        if fullpath.exists():
            fullpath = fullpath.with_name(f"{stem} ({i}){path.suffix}")
        else:
            return fullpath
```

Reply:
Avoid `exists()`, prefer to be more explicit with `is_dir()` and `is_file()`.

Reply:
No `exists`is the correct one here.
If there is a folder with that name I will be unable to create a file with the same name.
Or are you saying instead of `if x.exists():` I should do `if x.is_dir() or x.is_file():`, not considering things like block devices,. sockets and stuff?

Reply:
btw why `itertools.count(2)`?